matter/wifi/wf200/efr_spi.c

```diff
@@ -103,47 +112,67 @@ sl_status_t sl_wfx_host_init_bus(void)
 
   MY_USART->ROUTEPEN = USART_ROUTEPEN_TXPEN | USART_ROUTEPEN_RXPEN | USART_ROUTEPEN_CLKPEN;
   GPIO_DriveStrengthSet(SL_WFX_HOST_PINOUT_SPI_CLK_PORT, gpioDriveStrengthStrongAlternateStrong);
-
-#elif defined(EFR32MG24) || defined(EFR32MG21) /* Series 2 */
-  /*
-     * Route USART0 RX, TX, and CLK to the specified pins.  Note that CS is
-     * not controlled by USART0 so there is no write to the corresponding
-     * USARTROUTE register to do this.
-     */
-  GPIO->USARTROUTE[SL_WFX_HOST_PINOUT_SPI_PERIPHERAL_NO].RXROUTE =
-    (SL_WFX_HOST_PINOUT_SPI_RX_PORT << _GPIO_USART_RXROUTE_PORT_SHIFT)
-    | (SL_WFX_HOST_PINOUT_SPI_RX_PIN << _GPIO_USART_RXROUTE_PIN_SHIFT);
-  GPIO->USARTROUTE[SL_WFX_HOST_PINOUT_SPI_PERIPHERAL_NO].TXROUTE =
-    (SL_WFX_HOST_PINOUT_SPI_TX_PORT << _GPIO_USART_TXROUTE_PORT_SHIFT)
-    | (SL_WFX_HOST_PINOUT_SPI_TX_PIN << _GPIO_USART_TXROUTE_PIN_SHIFT);
-  GPIO->USARTROUTE[SL_WFX_HOST_PINOUT_SPI_PERIPHERAL_NO].CLKROUTE =
-    (SL_WFX_HOST_PINOUT_SPI_CLK_PORT << _GPIO_USART_CLKROUTE_PORT_SHIFT)
-    | (SL_WFX_HOST_PINOUT_SPI_CLK_PIN << _GPIO_USART_CLKROUTE_PIN_SHIFT);
-  GPIO->USARTROUTE[SL_WFX_HOST_PINOUT_SPI_PERIPHERAL_NO].CSROUTE =
-    (SL_WFX_HOST_PINOUT_SPI_CS_PORT << _GPIO_USART_CSROUTE_PORT_SHIFT)
-    | (SL_WFX_HOST_PINOUT_SPI_CS_PIN << _GPIO_USART_CSROUTE_PIN_SHIFT);
-
-  // Enable USART interface pins
-  GPIO->USARTROUTE[SL_WFX_HOST_PINOUT_SPI_PERIPHERAL_NO].ROUTEEN = GPIO_USART_ROUTEEN_RXPEN | // MISO
-                                                                   GPIO_USART_ROUTEEN_TXPEN | // MOSI
-                                                                   GPIO_USART_ROUTEEN_CLKPEN | GPIO_USART_ROUTEEN_CSPEN;
-
-#else
-#error "EFR32 type not supported"
-#endif
-  /* Configure CS pin as output and drive strength to inactive high */
+    /* Configure CS pin as output and drive strength to inactive high */
   GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_CS_PORT, SL_WFX_HOST_PINOUT_SPI_CS_PIN, gpioModePushPull, 1);
   GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_TX_PORT, SL_WFX_HOST_PINOUT_SPI_TX_PIN, gpioModePushPull, 0);
   GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_RX_PORT, SL_WFX_HOST_PINOUT_SPI_RX_PIN, gpioModeInput, 0);
   GPIO_PinModeSet(SL_WFX_HOST_PINOUT_SPI_CLK_PORT, SL_WFX_HOST_PINOUT_SPI_CLK_PIN, gpioModePushPull, 0);
+  
+#elif defined(EFR32MG24)  /* Series 2 */
+  // Initialize and enable the USART
+  USART_InitSync_TypeDef usartInit = USART_INITSYNC_DEFAULT;
+
+  MY_USART_CLOCK     = cmuClock_USART0;
+  MY_USART_TX_SIGNAL = dmadrvPeripheralSignal_USART0_TXBL;
+  MY_USART_RX_SIGNAL = dmadrvPeripheralSignal_USART0_RXDATAV;
+
+  spi_enabled = true;
+  dummy_tx_data = 0;
+  usartInit.baudrate = 36000000u;
+  usartInit.msbf = true;
+
+  CMU_ClockEnable(cmuClock_GPIO, true);
+  CMU_ClockEnable(MY_USART_CLOCK, true);
+  USART_InitSync(MY_USART, &usartInit);
```

Comment:
This goes again agaisnt the streamlining of the platform code and should be move to the CSA repo. It duplicates the SPI initialization and might cause some problem with other peripherals. Moreover this is for the WF200 which is really limited flash wise. So the use of an external spi flash is a strong possibility here.

Reply:
This section is the same as above for MG12. Why copy the code and place them in two #if def is the sequence init sequence is exactly the same.

Reply:
Modified the code as per the suggestion.